src/controller/java/src/chip/tlv/TlvReader.kt

```diff
@@ -355,6 +355,14 @@ class TlvReader(bytes: ByteArray) : Iterable<Element> {
     return bytes.size == index
   }
 
+  private fun Element.encodedLength() =
+    Tag.encode(value.toType().encode(), tag).size + value.encode().size
+
+  /** Returns true if this reader is currently pointing to the last element in the TLV data. */
+  fun isLastElement(): Boolean {
```

Comment:
can we use `=` here since function is trivial?

Reply:
This also seems to rely that decoding and re-encoding is the same size. Likely correct, but needs mental processing for this.

How about using the same logic as in peekElement (i.e. remembering the index) and checking if we reached the end index (i.e. `index >= bytes.size`)?

Reply:
This also seems to throw if you already read the entire tlv ... I would expect isLastElement to return true if the TLV is pointing to the last element and false otherwise. Instead this also has the case of `will throw if reader points to the end/is exhausted` which seems inconvenient.

What is the usage of this method? Why don't we have a `wasLastElement` maybe which is trivial?